packages/core-data/src/selectors.ts

```diff
@@ -71,31 +63,16 @@ interface UndoState extends Array< Object > {
 }
 
 interface UserState {
-	queries: Record< string, GenericRecordKey[] >;
-	byId: Record< GenericRecordKey, User< 'edit' > >;
+	queries: Record< string, EntityRecordKey[] >;
+	byId: Record< EntityRecordKey, User< 'edit' > >;
 }
 
-type GenericRecordKey = number | string;
-type EntityRecord = any;
 type Optional< T > = T | undefined;
 
 /**
  * HTTP Query parameters sent with the API request to fetch the entity records.
  */
-export type EntityQuery<
-	C extends Context,
-	WithFields extends boolean = true
-> = Omit< Record< string, any >, '_fields' > & {
-	context?: C;
-} & ( WithFields extends true
-		? {
-				/**
-				 * The requested fields. If specified, the REST API will remove from the response
-				 * any fields not on that list.
-				 */
-				_fields: string[];
-		  }
-		: {} );
+type EntityQuery = Record< string, any >;
```

Comment:
Would it still be possible to rename it to `EntityHttpQuery` so it's easier to follow?